### PR TITLE
[MySQL] Add SSL Support

### PR DESCRIPTION
**Pull request by ghost**
## Overview

Add SSL support.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

-

### CODE REVIEW

File: packages/mysql/manifest.yml [hunk 1]
```diff
@@ -55,6 +55,35 @@ policy_templates:
             title: Password
             secret: true
             default: test
+          - name: ssl
```
Comment (aliabbas-elastic) [inline packages/mysql/manifest.yml:58]: @gpop63 I don't see this being handled in the `stream.yml.hbs` file? [Example](https://github.com/elastic/integrations/pull/9414/files#diff-22dbe6efb734586cb633ae9440da040605455454ef56a0c3b0b2ac027f636fc7R9)

File: packages/mysql/manifest.yml [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
+version: "1.20.0"
 description: Collect logs and metrics from MySQL servers with Elastic Agent.
 type: integration
 categories:
   - datastore
   - observability
 conditions:
   kibana:
-    version: "^8.12.0"
+    version: "^8.14.0"
```
Comment (aliabbas-elastic) [inline packages/mysql/manifest.yml:12]: Any reason for bumping kibana version for this SSL enhancement?

**Review by ishleenk17 — COMMENTED** [review]
@andrewkroh : 
We are adding SSL support to one of our Integrations here. 
Has a proposal been formulated regarding the division of this, as was discussed in the meeting?
We can start implementing it from this Integration.

**Review by ishleenk17 — APPROVED** [review]
Looks good!
Please see why the CI is failing.

**Comment by ishleenk17** [comment]
@gpop63 : Can we resolve the CI issue and conclude on this. 
So that we are ready to go when 8.14 is released

**Comment by gpop63** [comment]
Locally all tests pass (pipeline, system) not sure what the CI doesn't like. Maybe the kibana bump to 8.14 might have something to do with it but locally I tested with `8.14-SNAPSHOT` and everything passes.

**Comment by lalit-satapathy** [comment]
> Locally all tests pass (pipeline, system) not sure what the CI doesn't like. Maybe the kibana bump to 8.14 might have something to do with it but locally I tested with `8.14-SNAPSHOT` and everything passes.
> 
> The strange thing is that commit [5da2b42](https://github.com/elastic/integrations/commit/5da2b425a070565ac71612f0568d38750179438c) was fine but commit [03c13b6](https://github.com/elastic/integrations/commit/03c13b64bcab72d4025cde6cecf0871c4db063a7) broke the CI.

@ishleenk17 how to un-gate this?

**Comment by jsoriano** [comment]
I would propose by now to provide a mapping for `host.mac` in the performance data stream, as is done in other data streams in the package.

**Comment by ishleenk17** [comment]
> I would propose by now to provide a mapping for `host.mac` in the performance data stream, as is done in other data streams in the package.

@jsoriano : I think the CI was [passing](https://github.com/elastic/integrations/commit/5da2b425a070565ac71612f0568d38750179438c) before and [failing](https://github.com/elastic/integrations/commit/03c13b64bcab72d4025cde6cecf0871c4db063a7) later because a different elastic package version was used.
Could you please tell me what checks the 0.99 elastic package is having that it started failing if host.mac was not detected in one of the datastreams? 
I would assume since ecs.yml are present, it would not use the component template?

**Comment by jsoriano** [comment]
> Could you please tell me what checks the 0.99 elastic package is having that it started failing if host.mac was not detected in one of the datastreams?

The issue seems to be related to the changes for `ecs@mappings` support (elastic/elastic-package#1711), effectively introduced in 0.99.
Since this version, packages that don't support versions of the stack older than 8.13, as is the case here, are validated against the whole ECS. This is because since this version all packages to use the `ecs@mappings` template.
This behaviour can be overriden for specific fields by defining them as in previous versions.

The failure here is produced by the `host.mac` field in the performance data stream. This field is defined in the other data streams of this package, but not in this one. So elastic-package is using the definition in ECS to validate this field. This definition includes the format seen in the error message: https://github.com/elastic/ecs/blob/f09fa4536990e90d5d3de9cff9d05985356ed7d9/generated/ecs/ecs_flat.yml#L5010

So at this point there are two ways to solve this issue:
* Adapt the format of these MAC addresses to the one expected by ECS, but this could be considered breaking as would duplicate values.
* Define the `host.mac` field in the performance data stream, so the definition in ECS is not used.

In previous versions of this package it looks like the `host.mac` field was not defined, what is also incorrect, I wonder why this was not causing issues before.

>  I would assume since ecs.yml are present, it would not use the component template?

The `ecs@mappings` component template is used in all packages starting on 8.13. The `ecs.yml` file is called like this by convention, it doesn't have any special meaning or trigger any special behaviour, nor the ECS fields need to be defined in this file.

**Comment by gpop63** [comment]
I tried removing `host.mac` from other data streams just for testing purposes as suggested by @ishleenk17  to see if tests would fail locally, but they still pass. Is there any way to replicate this locally? I used elastic-package `v0.100.0` and stack version `8.14-SNAPSHOT`.

**Comment by jsoriano** [comment]
> I wonder why this was not causing issues before.

Ok, the lack of definition was not an issue before because we skip validation of some fields that don't have a mapping for legacy reasons, what includes `host.*`, here: https://github.com/elastic/elastic-package/blob/db4c611d7d8e24e32261c2dc9d73ba77373078b0/internal/fields/validate.go#L582

> I tried removing `host.mac` from other data streams just for testing purposes as suggested by @ishleenk17 to see if tests would fail locally, but they still pass.

The `galera_status` data stream tests pass because it uses a different format for mac addresses, that seems to comply with ECS definition (`02-42-AC-1C-00-07`). And the `status` data stream doesn't seem to collect the `host.mac` field, at least not with the current test files.

> Is there any way to replicate this locally?

I can replicate this locally by running the static tests:
```
go run github.com/elastic/elastic-package stack up -v --version 8.14.0-SNAPSHOT -d
go run github.com/elastic/elastic-package test static -v
```

And I can confirm that the issue gets solved adding a mapping for `host.mac`.

**Comment by ishleenk17** [comment]
Thanks @jsoriano for the explanations. 
@gpop63 : Can you please check if we can replicate the issue with static test.
Lets add host.mac the way it is present in other datastreams to unblock ourselves.

**Pull request merged**